jetty-core/jetty-websocket/jetty-websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/messages/DispatchedMessageSink.java

```diff
@@ -48,8 +50,10 @@
 public abstract class DispatchedMessageSink extends AbstractMessageSink
 {
     private final Executor executor;
+    private final AtomicBoolean wasCallbackFailed = new AtomicBoolean(false);
```

Comment:
This is kind of a strange usage of an atomic here as your are not doing CaS operations on it.
Why not instead have a AtomicMarkedReference<Throwable>, with the mark indicating completion and the reference any failure?    It is a little bit more code, but will be easier reading.  It also will protect against multiple calls to succeeded and failed

Reply:
Not really sure how you would use an `AtomicMarkedReference` for this.

We just want to know if any of the callbacks from the message were failed. What would we store as the reference?

Reply:
hmmmm ok